### PR TITLE
[script] [equipmanager] [common] Add 'needs_unloading' property to DRC::Item class

### PR DESCRIPTION
**Pull request by KatoakDR**
### Background
* In the Lich discord, [VTCifer](https://discord.com/channels/745675889622384681/745696628714897508/857611999471075348) informed us that loaded bows, slings, and crossbows can be tied to some [tie items](https://elanthipedia.play.net/Category:Tie_items) and don't need to be unloaded

### Changes
* ~When deciding if need to unload weapon before stowing it, skip unloading if `item.tie_to` is defined~
* Add new gear item property `:needs_unloading: boolean` to players can express if they want their ranged weapon to be unloaded before being put away
* Moved logic for inferring if an item noun is a ranged weapon or not to `DRC::Item` class so that the logic can be used outside of `equipmanager`
* Update `DRC::Item.new` constructor to default `:ranged => nil` so that we treat true/false as the given values from the player, and if nil then we infer from the item noun. This was necessary so that players could say `:ranged => false` and their weapon would not default to trying to be unloaded if not using a tie item.

## Tests

<details>
<summary>Prove can tie loaded weapon to a tie item</summary>

```
> glance

You glance down to see a tenebrous demonbone-backed assassin's longbow adorned with serrated tips in your right hand and a lemon-striped arrow in your left hand.

> load longbow
Your demonscale shield makes the task more difficult.
You load the assassin's longbow with a lemon-striped arrow in your hand.
Roundtime 5 sec.

> look on longbow
On the assassin's longbow you see a lemon-striped arrow.

> put longbow in lootsack
You should unload the assassin's longbow first.

> tie longbow to lootsack
You attach a tenebrous demonbone-backed assassin's longbow adorned with serrated tips to a strap on your thornweave lootsack.

> look on longbow on lootsack
On the assassin's longbow you see a lemon-striped arrow.

> untie longbow from lootsack 
You untie a tenebrous demonbone-backed assassin's longbow adorned with serrated tips from the straps of your thornweave lootsack.

> load
Your assassin's longbow is already loaded with a lemon-striped arrow.
```
</details>

<details>
<summary>stow_weapon(item)</summary>

GIVEN:
```yaml
gear:
  - :adjective: assassin's
    :name: longbow
    :needs_unloading: false
    :tie_to: thornweave lootsack

  - :adjective: leather
    :name: sling
    :container: hitman backpack
```

THEN:
```
> ,e em = EquipmentManager.new ; echo em.stow_weapon("assassin's.longbow")

--- Lich: exec1 active.
[exec1]>tie my assassin's.longbow to my thornweave lootsack
You attach a tenebrous demonbone-backed assassin's longbow adorned with serrated tips to a strap on your thornweave lootsack.
--- Lich: exec1 has exited.

> ,e em = EquipmentManager.new ; echo em.stow_weapon("leather sling")

--- Lich: exec1 active.
[exec1]>unload my leather.sling
You unload the sling.
Roundtime: 1 seconds.
 
[exec1]>stow left
You put your rock in your black quiver.

[exec1]>put my leather.sling in my hitman.backpack
You put your sling in your hitman backpack.
--- Lich: exec1 has exited.
```

### CODE REVIEW

**Comment by KatoakDR** [comment]
In a follow up conversation with [VTCifer](https://discord.com/channels/745675889622384681/745696628714897508/858340557687357480), it was clarified that not all tie items support this mechanic of holding loaded weapons.

I'm going to refactor the logic to introduce a new item property `needs_unloading: boolean` so people can explicitly denote if they want their ranged weapons unloaded before being put away.

General idea will be if your yaml says `needs_unloading: true|false` then that is what the value will be; if it's null then it'll fall back infer based on if the weapon is a ranged weapon or not

File: equipmanager.lic [hunk 1]
```diff
@@ -18,7 +18,7 @@ class EquipmentManager
...
     @sort_head = settings.sort_auto_head
-    @items = settings.gear.map { |item| Item.new(name: item[:name], leather: item[:is_leather], hinders_locks: item[:hinders_lockpicking], worn: item[:is_worn], swappable: item[:swappable], tie_to: item[:tie_to], adjective: item[:adjective], bound: item[:bound], wield: item[:wield], transforms_to: item[:transforms_to], transform_verb: item[:transform_verb], transform_text: item[:transform_text], lodges: item[:lodges], ranged: item[:ranged], skip_repair: item[:skip_repair], container: item[:container]) }
+    @items = settings.gear.map { |item| Item.new(name: item[:name], leather: item[:is_leather], hinders_locks: item[:hinders_lockpicking], worn: item[:is_worn], swappable: item[:swappable], tie_to: item[:tie_to], adjective: item[:adjective], bound: item[:bound], wield: item[:wield], transforms_to: item[:transforms_to], transform_verb: item[:transform_verb], transform_text: item[:transform_text], lodges: item[:lodges], ranged: item[:ranged], needs_unloading: item[:needs_unloading], skip_repair: item[:skip_repair], container: item[:container]) }
```
Comment (wstampley) [inline equipmanager.lic:21]: Might be nice to move the mapping of the YAML property names to the Item property names into something like 'fromYAML' like your 'from_text' method, just to centralize and clean that up.

Comment (KatoakDR) [reply equipmanager.lic:21]: Yes, that'd be great! Since there's a few PRs in flight at the moment, I could take that on once the dust settles. Thanks

**Review by wstampley — APPROVED** [review]

**Comment by KatoakDR** [comment]
@rpherbig this is ready for your final review, thanks!

**Pull request merged**